### PR TITLE
fix(v3): Restore accessibility to all profiles when default profile has APIML token authentication

### PR DESCRIPTION
**Pull request by likhithanimma1**
## Proposed changes

This is a port of #2735 for the next branch.
## Release Notes

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by traeok — APPROVED** [review]

**Review by traeok — APPROVED** [review]

**Review by traeok — CHANGES_REQUESTED** [review]
Please see [this comment](https://github.com/zowe/vscode-extension-for-zowe/pull/2735#issuecomment-1966695064) on the v2 PR 😅

File: packages/zowe-explorer-api/src/profiles/ProfilesCache.ts
Comment (t1m0thyj) [inline packages/zowe-explorer-api/src/profiles/ProfilesCache.ts:455]: I think adding the check for user and password is good 👍 

Could we restore the old logic for port on lines 455 and 458?

(same comment as on the v2 version of this PR 😋)

**Review by traeok — APPROVED** [review]

**Review by t1m0thyj — DISMISSED** [review]

**Review by t1m0thyj — COMMENTED** [review]
LGTM, will approve once merge conflicts are resolved

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @likhithanimma1!

**Review by traeok — APPROVED** [review]

**Pull request merged**
